src/middlewared/middlewared/alembic/versions/12.0/2020-05-08_15-18_birthday.py

```diff
@@ -0,0 +1,34 @@
+"""empty message
+
+Revision ID: a3ac49efb063
+Revises: 43779dce3a07
+Create Date: 2020-05-08 15:18:35.333997+00:00
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from datetime import datetime
+
+# revision identifiers, used by Alembic.
+revision = 'a3ac49efb063'
+down_revision = '43779dce3a07'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('system_settings', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('stg_birthday', sa.DateTime(), nullable=True))
+
+    op.execute("UPDATE system_settings SET stg_birthday = '" + str(datetime(1970, 1, 1, 0, 0, 0)) + "'")
```

Comment:
This will set 1970 even for new installs, wont it @themylogin 

Can you explain to him how to make it behave properly, please?

Reply:
@bastiao do not set any default value in migration. Put some flag (e.g. `set_birthday: true`) into `system_keyvalue` table in installer (`src/freenas-installer/etc/installer12.sh`). Pop this flag in middleware and set birthday accordingly.

Reply:
@themylogin  dont we set any environment var when doing install vs upgrade to make this simpler/easier?

Reply:
@william-gr forgot about that, updated my comment

Reply:
@themylogin why put anything under system_keyvalue?

Why not just update the stg_birthday value based on that env var, directly?

Reply:
@william-gr update how? Set to `0000-00-00`?

Reply:
His intent was to set datetime(1970, 1, 1, 0, 0, 0) if its an upgrade

so only run that UPDATE if `os.environ["FREENAS_INSTALL"] != "yes"`

Reply:
Do we want to have 1970 for upgrades?

Reply:
@william-gr updated my comment once more

@bastiao please only do it if `os.environ.get("FREENAS_INSTALL") != "yes"`